### PR TITLE
Chore/update 2312

### LINKED ISSUE #1509: Build docker iamge from scratch failed due to mysql pub key expired.
Error message:
```
Err:1 https://repo.mysql.com/apt/debian bullseye InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B7B3B788A8D3785C
Get:5 http://deb.debian.org/debian bullseye/main amd64 Packages [8062 kB]
Get:6 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [263 kB]
Get:7 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [18.8 kB]
Reading package lists...
W: GPG error: https://repo.mysql.com/apt/debian bullseye InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B7B3B788A8D3785C
E: The repository 'https://repo.mysql.com/apt/debian bullseye InRelease' is not signed.
```
The latestest key of mysql repo in [keyserver](https://keyserver.ubuntu.com/pks/lookup?search=mysql-build%40oss.oracle.com&fingerprint=on&op=index) is:

https://keyserver.ubuntu.com/pks/lookup?op=get&search=0xbca43417c3b485dd128ec6d4b7b3b788a8d3785c

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/SYNC-4055)

### PR DESCRIPTION
**Pull request by jrconlin**
## Description

Update for Dec-2023

* This updates or locks the various dependencies.
* This includes a fix for  #1509, which now pulls the latest MySQL Public key from the MySQL repo instead of loading it directly. It also uses the latest recommended way to include key files under Ubuntu. 
* It also fixes a SQL error in one of the migration files.

## Testing

Tests included

## Issue(s)

Closes #1509, SYNC-4055

### CODE REVIEW

**Review by pjenvey — COMMENTED** [review]
just a question about the sql formatting ^

File: syncstorage-mysql/migrations/2018-08-28-010336_init/up.sql [hunk 1]
```diff
@@ -1,69 +1,55 @@
 -- /*!40100 DEFAULT CHARACTER SET latin1 */
-
 -- DROP TABLE IF EXISTS `bso`;
 -- XXX: bsov1, etc
 CREATE TABLE IF NOT EXISTS `bso`(
-    `user_id` INT                           NOT NULL,
-    `collection_id` INT                     NOT NULL,
-    `id` VARCHAR(64)                        NOT NULL,
-
+    `user_id` INT NOT NULL,
```
Comment (pjenvey) [inline syncstorage-mysql/migrations/2018-08-28-010336_init/up.sql:5]: nit: did the IDE reformat this? I've tried to keep this styled after https://www.sqlstyle.guide/#create-syntax

Comment (jrconlin) [reply syncstorage-mysql/migrations/2018-08-28-010336_init/up.sql:5]: Yeah, it must have gotten auto-formatted.

**Review by pjenvey — APPROVED** [review]

**Comment by pjenvey** [comment]
I thought syncstorage already went through a config upgrade that changed all of its separators -- but maybe that was just a [change from one of the refactors](https://github.com/mozilla-services/cloudops-infra/commit/ce90139ad67c911e71e155222c637804d07be45b) (maybe I'm thinking of merino that had to change the separator at one point).

Anyway I *believe* there's actually a way to force the newer config crate to keep the old separator style for backwards compatibility -- it has a "separator" and I think also a "prefix_separator" you can set to do this. I think we just missed when we were forced to change all the settings for I believe merino.

**Comment by jrconlin** [comment]
Oh, huh. I think I remember you talking about that before, but I can't remember the details. 

I'll make a note about it for the next time.

**Pull request merged**